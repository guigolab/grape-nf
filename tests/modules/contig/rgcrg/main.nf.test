nextflow_process {

    name "Test Process contig"
    script "modules/contig/rgcrg/main.nf"
    process "contig"

    tag "module"

    setup {
        run("fastaIndex") {
            script = "modules/fastaIndex/samtools/main.nf"
            process {
                """
                input[0] = "${baseDir}/data/genome.fa.gz"
                """
            }
        }
    }

    test("Should produce BED file with contigs") {

        when {
            process {
                """
                input[0] = fastaIndex.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE"]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
