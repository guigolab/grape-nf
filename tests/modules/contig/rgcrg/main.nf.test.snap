{
    "Should produce BED file with contigs": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.contigs.bed:md5,89f19344b1f1a346a105f2b3f9443f9e",
                        "bed",
                        "Contigs",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T18:57:32.893994"
    }
}